apps/hash-graph/docker/Dockerfile

```diff
@@ -5,36 +5,42 @@ ENV RUSTUP_HOME=/usr/local/rustup \
     PATH=:$PATH:/usr/local/cargo/bin
 
 COPY rust-toolchain.toml .
-RUN apk add --no-cache gcc musl-dev && \
+RUN apk add --no-cache gcc musl-dev bash && \
     wget -q -O- https://sh.rustup.rs | sh -s -- -y --default-toolchain none --profile minimal && \
     rustup show
 
+SHELL ["bash", "-c"]
+
+
 FROM rust AS builder
 WORKDIR /usr/local/src/hash-graph
 COPY . .
 ARG PROFILE=production
+ARG TYPE_FETCHER=yes
 
 # To be removed once https://github.com/open-telemetry/opentelemetry-rust/issues/934 is sorted
-RUN apk add --no-cache make openssl-dev protobuf-dev
+RUN apk add --no-cache make protobuf-dev
 
 RUN --mount=type=cache,target=/usr/local/cargo/registry \
     --mount=type=cache,target=/usr/local/cargo/git \
     --mount=type=cache,target=/usr/local/src/hash-graph/target \
-    cargo install --path bin/cli --profile $PROFILE --locked
-
-# Replace with `FROM scratch` when a health check command was added to hash-graph
-FROM alpine:3.17 AS runtime
+    if [[ ${TYPE_FETCHER^^} == Y* || ${TYPE_FETCHER^^} == T* || $TYPE_FETCHER == 1 ]] ; then  \
+      cargo install --path bin/cli --profile $PROFILE --features type-fetcher --locked; \
+    else \
+      cargo install --path bin/cli --profile $PROFILE --locked;  \
+    fi
 
-RUN addgroup -S appgroup && adduser -S appuser -G appgroup
+RUN mkdir -p /out/etc/ \
+ && echo 'nobody:x:65534:65534:nobody:/:' > /out/etc/passwd \
+ && echo 'nobody:x:65534:' > /out/etc/group \
+ && install -d -m 0755 -o nobody -g nobody /out/log \
```

Comment:
I suppose this is good to have in case a volume isn't mounted. Although `read_only: true` likely would make the binary fail

Reply:
`read_only: true` is only specified in `docker-compose`, where we also specify the volume to mount. Without a compose-environment, you can still write to the image.